src/analysis/processing/qgsalgorithmwritevectortiles.cpp

```diff
@@ -0,0 +1,195 @@
+      if ( layerVariantMap.contains( "minZoom" ) )
+        writerLayer.setMinZoom( layerVariantMap["minZoom"].toInt() );
+      if ( layerVariantMap.contains( "maxZoom" ) )
+        writerLayer.setMaxZoom( layerVariantMap["maxZoom"].toInt() );
+      if ( layerVariantMap.contains( "layerName" ) )
+        writerLayer.setLayerName( layerVariantMap["layerName"].toString() );
+      layers << writerLayer;
+    }
+    return layers;
+  }
+
+};
+
+QString QgsWriteVectorTilesAlgorithm::name() const
+{
+  return QStringLiteral( "writevectortiles" );
+}
+
+QString QgsWriteVectorTilesAlgorithm::displayName() const
+{
+  return QObject::tr( "Write Vector Tiles" );
+}
+
+QString QgsWriteVectorTilesAlgorithm::group() const
+{
+  return QObject::tr( "Vector tiles" );
+}
+
+QString QgsWriteVectorTilesAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectortiles" );
+}
+
+QString QgsWriteVectorTilesAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "blah blah" );  // TODO
+}
+
+QgsProcessingAlgorithm *QgsWriteVectorTilesAlgorithm::createInstance() const
+{
+  return new QgsWriteVectorTilesAlgorithm();
+}
+
+void QgsWriteVectorTilesAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterVectorTileWriterLayers( "LAYERS", QObject::tr("Input layers") ) );
+
+  addParameter( new QgsProcessingParameterString( "XYZ_TEMPLATE", QObject::tr("File template"), "/home/qgis/{z}/{x}/{y}.pbf" ) );
+  // TODO maybe use this:
+  // addParameter( new QgsProcessingParameterFileDestination( QStringLiteral( "OUTPUT" ), QObject::tr( "Destination GeoPackage" ), QObject::tr( "GeoPackage files (*.gpkg)" ) ) );
+
+  addParameter( new QgsProcessingParameterNumber( "MIN_ZOOM", QObject::tr("Minimum zoom level"), QgsProcessingParameterNumber::Integer, 0, false, 0, 24 ) );
+  addParameter( new QgsProcessingParameterNumber( "MAX_ZOOM", QObject::tr("Maximum zoom level"), QgsProcessingParameterNumber::Integer, 3, false, 0, 24 ) );
+
+  // optional extent
+  addParameter( new QgsProcessingParameterExtent( "EXTENT", QObject::tr("Extent"), QVariant(), true ) );
+
+  // TODO: optional metadata (only for MBTiles? and with concrete values rather than single QVariantMap?)
+}
+
+QVariantMap QgsWriteVectorTilesAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  int minZoom = parameterAsInt( parameters, QStringLiteral( "MIN_ZOOM" ), context );
+  int maxZoom = parameterAsInt( parameters, QStringLiteral( "MAX_ZOOM" ), context );
+
+  // prepare output URI
+  QString xyzTemplate = parameterAsString( parameters, QStringLiteral( "XYZ_TEMPLATE" ), context );
+  QgsDataSourceUri dsUri;
+  dsUri.setParam( "type", "xyz" );
+  dsUri.setParam( "url", QUrl::fromLocalFile( xyzTemplate ).toString() );
+  QString uri = dsUri.encodedUri();
+
+  QVariant layersVariant = parameters.value( parameterDefinition( QStringLiteral( "LAYERS" ) )->name() );
```

Comment:
```suggestion
  QVariant layersVariant = parameters.value( QStringLiteral( "LAYERS" ) );
```

save a little bit of unecessary work :laughing:

Reply:
Actually this is what I had initially. But then I have realized that parameterDefinition() uses also case insensitive matching of parameter names, and all other algorithms seem to use that function indirectly, so I thought that should be supported here too?

Reply:
Hm - good point!